storage/interface.go

```diff
@@ -282,7 +282,18 @@ type HistogramAppender interface {
 	// For efficiency reasons, the histogram is passed as a
 	// pointer. AppendHistogram won't mutate the histogram, but in turn
 	// depends on the caller to not mutate it either.
-	AppendHistogram(ref SeriesRef, l labels.Labels, t int64, h *histogram.Histogram, fh *histogram.FloatHistogram) (SeriesRef, error)
+	//
+	// By default, the storage should assume the histogram to be a counter histogram
+	// unless the hints say it is a gauge histogram.
+	AppendHistogram(ref SeriesRef, l labels.Labels, t int64, h *histogram.Histogram, fh *histogram.FloatHistogram, hints *AppendHints) (SeriesRef, error)
```

Comment:
Isn't it a bit of an overhead to pass the AppendHints as a pointer? Even if we add more hints to the struct later, it might be leaner to not use a pointer. Which would also avoid ownership problems (if someone recycles the AppendHints).

Reply:
Why adding a pointer adds more overhead?

Reply:
It means an additional step of de-referencing when accessing the value (only matters in very hot code paths, of course). And it might confuse the escape analysis, maybe triggering the compiler to do a heap allocation where we don't need it.